app.js

```diff
@@ -282,6 +293,54 @@ mongoose.connect(settings.database.uri, function(err) {
             return process.exit();
         }
 
+        console.log('Starting app.');
         startApp();
     });
+
+}
+
+const Kefir = require('kefir');
+const interval = 2000;
+const MongoClient = require('mongodb').MongoClient;
+
+function tryConnect(callback) {
+    console.log('Connecting to database...');
+    console.log(new Date());
+    console.log(`connection is ${settings.database.uri}`)
+    MongoClient.connect(settings.database.uri, callback);
+
+
+    mongoose.connection.on('error', function (err) {
+        console.log('connection on error ${err}');
+        //callback(err);
+    });
+
+    mongoose.connection.on('disconnected', function (err) {
+        console.log(`disconnected from database error : ${err}`);
+          //callback("error");
+    });
+    //if (retryNum < 3)
+    //  throw new Error('number of connection bigger the 3');
+}
+
+  let startStream = Kefir.repeat((e)=>{
```

Comment:
This would just run `tryConnect` sequentially, regardless of the connection status. It's basically equivalent to:

```
Kefir.sequentially(1000, _.range(5)).flatMap(()=> Kefir.fromNodeCallback(tryConnect))
```

Try this instead:
```javascript
let mongoConnectionProperty = Kefir.repeat(()=>{
    return Kefir.concat([
        Kefir
            .fromPromise(MongoClient.connect(settings.database.uri))
            .flatMap((db)=>{
                return Kefir.merge([
                    Kefir.constant(db),
                    Kefir.stream(_.noop)
                ]).takeUntilBy(Kefir.fromEvents(db, 'close'));
            }),
        Kefir.later(2000).flatMap(Kefir.never)
    ])
}).toProperty();
```
It should give you a steady property containing the latest connected `db` instance.

To use it, you can listen to the property values (i.e. `mongoConnectionProperty.onValue()`) or combine it with another stream (probably "silently") (i.e. `Kefir.combine([anotherStream], [mongoConnectionProperty])`